src/hooks/useCountdownTimer/useCountdownTimer.tsx

```diff
@@ -0,0 +1,27 @@
+import { useCallback, useEffect, useState } from 'react'
+
+export const useCountdownTimer = (initialSeconds: number) => {
+  const [secondsRemaining, setSecondsRemaining] = useState(initialSeconds)
+  const [progress, setProgress] = useState(100)
+
+  const reset = useCallback(() => {
+    setSecondsRemaining(initialSeconds)
+    setProgress(100)
+  }, [initialSeconds])
+
+  useEffect(() => {
+    const intervalId = setInterval(() => {
+      if (secondsRemaining > 0) {
+        const newProgress = (secondsRemaining / initialSeconds) * 100
+        setProgress(newProgress)
+        setSecondsRemaining(secondsRemaining - 1)
```

Comment:
this introduces an infinite loop because it's setting state thats being observed. You'll need to use a ref in this case

Reply:
We aren't using this, going to remove it.